pkg/loop/internal/relayer/pluginprovider/ext/ccipocr3/chainaccessor.go

```diff
@@ -77,16 +88,19 @@ func (c *chainAccessorClient) GetChainFeeComponents(ctx context.Context) (ccipoc
 	}, nil
 }
 
-func (c *chainAccessorClient) Sync(ctx context.Context, contractName string, contractAddress ccipocr3.UnknownAddress) error {
-	_, err := c.grpc.Sync(ctx, &ccipocr3pb.SyncRequest{
-		ContractName:    contractName,
-		ContractAddress: contractAddress,
-	})
+func (c *ChainAccessorClient) Sync(ctx context.Context, contractName string, contractAddress ccipocr3.UnknownAddress) error {
+	req := &ccipocr3pb.SyncRequest{ContractName: contractName, ContractAddress: contractAddress}
+	_, err := c.grpc.Sync(ctx, req)
+	if err != nil {
+		c.mu.Lock()
+		c.syncs = append(c.syncs, req) // TODO dedupe?
```

Comment:
Probably just using a map is simplest. I was trying to maintain the protobuf caching but that is nbd.